sdv/datasets/demo.py

```diff
@@ -0,0 +1,122 @@
+"""Methods to load datasets."""
+
+import io
+import logging
+import os
+import shutil
+import tempfile
+import urllib.request
+from zipfile import ZipFile
+
+import pandas as pd
+
+from sdv.metadata.multi_table import MultiTableMetadata
+from sdv.metadata.single_table import SingleTableMetadata
+
+LOGGER = logging.getLogger(__name__)
+BUCKET_URL = 'https://sdv-demo-datasets.s3.amazonaws.com'
+METADATA_FILENAME = 'metadata.json'
+
+
+def _get_dataset_url(modality, dataset_name):
+    return os.path.join(BUCKET_URL, modality.upper(), dataset_name + '.zip')
+
+
+def _validate_args_download_demo(modality, output_folder_name):
+    possible_modalities = ['single_table', 'multi_table', 'sequential']
+    if modality not in possible_modalities:
+        raise ValueError(f"'modality' must be in {possible_modalities}.")
+
+    if output_folder_name and os.path.exists(output_folder_name):
+        raise ValueError(
+            f"Folder '{output_folder_name}' already exists. Please specify a different name "
+            "or use 'load_csvs' to load from an existing folder."
+        )
+
+
+def _download(modality, dataset_name, output_folder_name):
+    dataset_url = _get_dataset_url(modality, dataset_name)
+    LOGGER.info(f'Downloading dataset {dataset_name} from {dataset_url}')
+    try:
+        response = urllib.request.urlopen(dataset_url)
+    except urllib.error.HTTPError:
+        raise ValueError(
+            f"Invalid dataset name '{dataset_name}'. "
+            "Use 'list_available_demos' to get a list of demo datasets."
```

Comment:
minor: I think we could add a line here saying "make sure you have the correct modality for the dataset name" or something like that